File: ml/config/languageConfig.py
# ml/config/languageConfig.py

"""
Language configuration for ML backend
Provides language mappings for OCR, TTS, and text analysis
"""

# Language configurations
LANGUAGES = {
    'en': {
        'code': 'en',
        'name': 'English',
        'tesseract': 'eng',
        'gtts': 'en',
        'script': 'Latin'
    },
    'hi': {
        'code': 'hi',
        'name': 'Hindi',
        'tesseract': 'hin',
        'gtts': 'hi',
        'script': 'Devanagari'
    },
    'kn': {
        'code': 'kn',
        'name': 'Kannada',
        'tesseract': 'kan',
        'gtts': 'kn',
        'script': 'Kannada'
    }
}

DEFAULT_LANGUAGE = 'en'


def get_language_config(language_code):
    """
    Get language configuration
    
    Args:
        language_code: Language code (en, hi, kn)
    
    Returns:
        dict: Language configuration
    """
    return LANGUAGES.get(language_code, LANGUAGES[DEFAULT_LANGUAGE])


def is_valid_language(language_code):
    """
    Check if language is supported
    
    Args:
        language_code: Language code to check
    
    Returns:
        bool: True if supported
    """
    return language_code in LANGUAGES


def get_tesseract_lang(language_code):
    """
    Get Tesseract language code
    
    Args:
        language_code: Language code
    
    Returns:
        str: Tesseract language code
    """
    config = get_language_config(language_code)
    return config['tesseract']


def get_gtts_lang(language_code):
    """
    Get gTTS language code
    
    Args:
        language_code: Language code
    
    Returns:
        str: gTTS language code
    """
    config = get_language_config(language_code)
    return config['gtts']


def get_script_name(language_code):
    """
    Get script name for language
    
    Args:
        language_code: Language code
    
    Returns:
        str: Script name
    """
    config = get_language_config(language_code)
    return config['script']


# Export all functions
__all__ = [
    'LANGUAGES',
    'DEFAULT_LANGUAGE',
    'get_language_config',
    'is_valid_language',
    'get_tesseract_lang',
    'get_gtts_lang',
    'get_script_name'
]

File: backend/config/languageConfig.js
// backend/config/languageConfig.js

const LANGUAGES = {
  en: {
    code: "en",
    name: "English",
    ocrLang: "eng",      // Tesseract language code
    nativeName: "English"
  },
  hi: {
    code: "hi",
    name: "Hindi",
    ocrLang: "hin",      // Tesseract language code for Hindi
    nativeName: "हिंदी"
  },
  kn: {
    code: "kn",
    name: "Kannada",
    ocrLang: "kan",      // Tesseract language code for Kannada
    nativeName: "ಕನ್ನಡ"
  }
};

const DEFAULT_LANGUAGE = "en";

function getLanguageConfig(languageCode) {
  return LANGUAGES[languageCode] || LANGUAGES[DEFAULT_LANGUAGE];
}

function isValidLanguage(languageCode) {
  return languageCode in LANGUAGES;
}

module.exports = {
  LANGUAGES,
  DEFAULT_LANGUAGE,
  getLanguageConfig,
  isValidLanguage,
};